scanapi/tree/request_node.py

```diff
@@ -155,16 +155,21 @@ def run(self):
         url = self.full_url_path
         console.print(f"\n- Making request {method} {url}", highlight=False)
 
-        session = session_with_retry(self.retry)
-        response = session.request(
-            method,
-            url,
-            headers=self.headers,
-            params=self.params,
-            json=self.body,
-            allow_redirects=False,
+        kwargs = {
+            "headers": self.headers,
+            "params": self.params,
+            "json": self.body,
+            "allow_redirects": False,
             **self.options,
-        )
+        }
+
+        for k, v in self.headers.items():
+            if k.lower() == "content-type" and v.lower() != "application/json":
+                kwargs["data"] = self.body
+                kwargs.pop("json", None)
```

Comment:
Since we know `"json"` will always be present, we can use `del` directly
```suggestion
                del kwargs["json"]
```

Reply:
So... maybe it's precious, but dict isn't case sensitive, then if an User set a key `Content-Type` in the request spec and `content-type` in the endpoint spec it will raises an error here.. uses `break` keyword solve this problem what u think? 🤔

Reply:
@camilamaia maybe we should use a no case sensitive dict to the headers? 🤔

Reply:
Good point! Since header names are case **in**sensitive according to the HTTP specification, I agree that we should make them also case **in**sensitive here 👍

Reply:
I'm sorry, I don't understand what this discussion has to do with the `kwargs.pop` vs `del` here?

Case-insensitive dict for the headers is a good idea yes, but maybe for another PR? It's already an existing issue: if somebody defines two headers with different cases then what's the behavior?

Reply:
@Crocmagnon i changed this code snipet, then this discussion is deprecated.
Also, about the case-insensitive dict i will do it in another PR 👀